.github/workflows/type-snapshot.yml

```diff
@@ -44,9 +44,41 @@ jobs:
           bazel build --disk_cache=~/bazel-disk-cache --strip=always --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --config=ci --config=release_linux //types:types
       - name: Check snapshot diff
         run: |
-          diff -r bazel-bin/types/definitions/ types/generated-snapshot
-      - name: How to fix
+          diff -r bazel-bin/types/definitions/ types/generated-snapshot > types.diff
+      - name: 'Put diff on the environment'
         if: failure()
+        id: types_diff
         run: |
-          echo "The generated output of @cloudflare/workers-types has been changed by this PR"
-          echo "If this is intentional, run 'bazel build //types && rm -rf types/generated-snapshot && cp -r bazel-bin/types/definitions types/generated-snapshot' to update the snapshot"
+          {
+            echo 'TYPES_DIFF<<EOF'
+            cat types.diff
+            echo EOF
+          } >> "$GITHUB_ENV"
+      - uses: actions/upload-artifact@v4
+        id: artifact-upload-step
+        if: failure()
+        with:
+          name: generated-snapshot
+          path: bazel-bin/types/definitions/
+      - name: 'Comment on PR with error details'
+        if: failure()
+        uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
We probably shouldn't have our CI depending on random unknown people's action helpers, unless we can lock it at a commit or something.